src/components/SortButton.tsx

```diff
@@ -0,0 +1,151 @@
+import React, { useState } from 'react'
+import { useSelector } from 'react-redux'
+import { css } from '../../styled-system/css'
+import theme from '../selectors/theme'
+import SortIcon from './icons/SortIcon'
+
+/**
+ * SortButton component.
+ * */
+const SortButton: React.FC<{ onSortChange: (sortOrder: 'alphabetical' | 'type') => void }> = ({ onSortChange }) => {
+  const isLightTheme = useSelector(state => theme(state) === 'Light')
+  const [isDropdownOpen, setDropdownOpen] = useState(false)
+  const [selectedSort, setSelectedSort] = useState<'alphabetical' | 'type'>('type')
+
+  const dropdownStyles = css({
+    opacity: 0,
+    visibility: 'hidden',
+    transition: 'opacity 0.3s ease-in-out, visibility 0.3s ease-in-out',
+    willChange: 'opacity, visibility',
+    position: 'absolute',
+    top: '100%',
+    right: 0,
+    backgroundColor: '{colors.bg}',
+    border: 'solid 1px {colors.gray50}',
+    borderRadius: '8px',
+    zIndex: 'modal',
+    padding: '1rem',
+    display: 'flex',
+    flexDirection: 'column',
+    gap: '0.5rem',
+  })
+
+  /**
+   * Fade in animation styles for the dropdown when it is visible.
+   * */
+  const dropdownVisibleStyles = css({
+    opacity: 1,
+    visibility: 'visible',
+  })
+
+  /**
+   * Handles the sort change.
+   */
+  const handleSortChange = (sortOrder: 'alphabetical' | 'type') => {
+    setSelectedSort(sortOrder)
+    onSortChange(sortOrder)
+    setDropdownOpen(false)
+  }
+
+  return (
+    <button
+      onClick={() => setDropdownOpen(!isDropdownOpen)}
+      className={css({
+        width: '45px',
+        border: 'solid 1px {colors.gray50}',
+        backgroundColor: '{colors.darkgray}',
+        borderRadius: '8px',
+        display: 'flex',
+        alignItems: 'center',
+        justifyContent: 'center',
+        position: 'relative',
+      })}
+    >
+      <SortIcon size={20} fill={isLightTheme ? '{colors.light}' : '{colors.fg}'} />
+      <div className={`${dropdownStyles} ${isDropdownOpen ? dropdownVisibleStyles : ''}`}>
```

Comment:
Use `css` or `cx` to merge Panda styles.

See: https://panda-css.com/docs/concepts/merging-styles